src/main/scala/scala/collection/immutable/LazyListExtensions.scala

```diff
@@ -0,0 +1,22 @@
+package scala.collection.immutable
+
+object LazyListExtensions {
+
+  implicit class LazyListCycle[T](ll: LazyList[T]) {
+    /**
+     * When called on a finite `LazyList`, returns a circular structure
+     * that endlessly repeats the elements in the input.
+     * The result is a true cycle occupying only constant memory.
+     */
+    def cycle: LazyList[T] =
+      if (ll.knownSize == 0) LazyList.empty
+      else LazyList.empty #::: {
```

Comment:
Just curious: why is this `LazyList.empty` required?

Reply:
it makes the rhs lazy

Reply:
It's awkward that all the usual ways of constructing `LazyList`s require strict knowledge of whether the list is empty or not. `LazyList.empty #:::` is a workaround for what is arguably a gap in the API. We discussed it previously here: https://github.com/scala/bug/issues/11984#issuecomment-625528485